components/kyma-environment-broker/internal/orchestration/kyma/manager.go

```diff
@@ -150,12 +152,18 @@ func (u *upgradeKymaManager) resolveOperations(o *internal.Orchestration, params
 		o.Description = fmt.Sprintf("Scheduled %d operations", len(runtimes))
 
 	} else {
-		// Resume processing of in progress upgrade operations after restart
+		// Resume processing of not finished upgrade operations after restart
 		var err error
-		result, _, _, err = u.operationStorage.ListUpgradeKymaOperationsByOrchestrationID(o.OrchestrationID, dbmodel.OperationFilter{States: []string{string(domain.InProgress)}})
+		inProgress, _, _, err := u.operationStorage.ListUpgradeKymaOperationsByOrchestrationID(o.OrchestrationID, dbmodel.OperationFilter{States: []string{orchestration.InProgress}})
 		if err != nil {
 			return result, err
 		}
+		pending, _, _, err := u.operationStorage.ListUpgradeKymaOperationsByOrchestrationID(o.OrchestrationID, dbmodel.OperationFilter{States: []string{orchestration.Pending}})
```

Comment:
You could merge these two list with passing `States: []string{orchestration.Pending, orchestration.InProgress}`

Reply:
OK, I used a single list as you suggested and then I split operations per pending and in-progress states in the separate logic because I want to reprocess already in-progress operations at first to keep the order of executing operations

Reply:
Ah, I you need to process them separately, you could also keep the two queries, it shouldn't be too much of a difference performance wise.

Reply:
it's quite readable so I'll leave it as it's now, also one query less is always better